airflow/providers/google/cloud/utils/credentials_provider.py

```diff
@@ -158,3 +161,43 @@ def provide_gcp_conn_and_credentials(
         key_file_path, scopes, project_id
     ):
         yield
+
+
+def get_credentials_and_project_id(
```

Comment:
This code is very similar to: https://github.com/apache/airflow/blob/master/airflow/providers/google/cloud/hooks/base.py#L172-L218
Can you remove code duplication, please?

Reply:
I refactor the code to move the whole logic in `base.py` into `credentials_provider.py`. Let me know how you think

Reply:
LGTM